### ISSUE TITLE
Rename haproxy job

### ISSUE THREAD

**Issue by hannayurkevich** [logsearch/logsearch-boshrelease#226]
Hi guys,

I think it makes sense to use more specific name for `haproxy` job. Something like e.g. `logsearch-haproxy`. Then use this specific name when parsing logs from the job:

`if [syslog_program] == "haproxy" {` //  **use more specific name here** 
 `<%= File.read('src/logstash-filters/snippets/haproxy.conf').gsub(/^/, '  ') %>`
`}`
(see https://github.com/logsearch/logsearch-boshrelease/blob/develop/src/logsearch-config/src/logstash-filters/default.conf.erb)

Why this is a problem? I can explain it on example of our project where we set both **logsearch-boshrelease + logsearch-for-cloudfoundry**. We also have a **haproxy** job.
And logs from our **haproxy** are parsed by the code I mentioned above. As the result we have no control on parsing of these logs and unable to customize the rules.

I think that name **haproxy** is too generic to use it as a discriminating value when defining a special parsing. Please rename the **haproxy** job in your code to make related special parsing more "specific".

**Comment by mrdavidlaing** [comment]
@hannayurkevich - we agree; haproxy is 
1. too implementation specific - it should be called `router`; and
2. too generic.

What do you think about renaming all the jobs to ensure they are easily identifiable as "logsearch" jobs?

eg:

```
ls-elasticsearch_data
ls-elasticsearch_master
ls-firehose-to-syslog
ls-ingestor
ls-kibana
ls-maintenance
ls-monitor
ls-parser
ls-queue
ls-router
```

**Comment by hannayurkevich** [comment]
@mrdavidlaing, this seems really reasonable to me. Thank you!

**Comment by kirederik** [comment]
We tried to rename all jobs to `ls-<job name>` and found a couple of issues:
1. Bosh tries to spin up the new job before tear down the old one, which causes ip conflicts.
2. Bosh doesn't treat the ls-jobs as "renamed jobs" but as new jobs, which will cause data loss (new disks will be attached to the ls-jobs).

It might be possible to rename the jobs without losing data, by using the `bosh attach disk` command.

After discussion, we agreed with only renaming the haproxy job to `ls-router`, fulfilling  the original issue.

**Comment by hannayurkevich** [comment]
@kirederik,

Could you please change parsing rules for `haproxy` as well to reflect new job name?

Thank you in advance.

**Comment by cromega** [comment]
@hannayurkevich, @mrdavidlaing: This issue made us realise something that may serve as an important input when we discuss how to configure the parsing rules. We have some discussions lined up about that.

Essentially, what you wanted, is to somehow make our default haproxy parsing rule to **not** process the logs coming from your haproxy, is that correct?

We think that there is no way to provide default filtering rules that are generic enough for every use case. Most likely we will change the parsing rules in the core boshrelease to do syslog parsing only and then leave the rest up to whoever is consuming the release.

What do you think about it?

**Comment by kirederik** [comment]
@hannayurkevich As part of 2cac0366ca3e5e5bd37417e9634f3902cee5ebe9, we moved the haproxy parsing rule to the cluster_monitor job. As a result, we will not apply the haproxy parsing rule on the regular cluster.

We also renamed the default syslog program name from `haproxy` to `ls-router`, allowing us to change the parsing rule. 

What do you think about this? Can we close this issue?

Thanks!

**Comment by hannayurkevich** [comment]
@kirederik,

Thank you for the changes! They fix the problem I described, yeah.

The is only one thing. It's can be confusing that you mix up "haproxy" and "ls-router" names.
I mean It would be great if you rename `haproxy.conf` (and corresponding test files),`[haproxy][*]` fields, `haproxy` tag etc. and use more specific names for them too (something related to **ls-router**). This doesn't actually affect parsing but can make the code more convenient and remove any confusion.

What do you think? Does it make sense?

**Comment by cromega** [comment]
@hannayurkevich: we thought about it but in the end ls-router is just haproxy, so what else to call a parsing rule that understand that log format if not haproxy?

**Comment by hannayurkevich** [comment]
@cromega Yeah, it sounds reasonable.

Guys, I would really appreciate it if you make a release including the fix you've made.

And I think the issue can be closed. Many thanks!

**Comment by mrdavidlaing** [comment]
@hannayurkevich - Glad we've reached a point that solves your issue.

I'll schedule a release to bosh.io for the end of this week.

**Comment by mrdavidlaing** [comment]
@hannayurkevich - This fix is included in `logsearch v202.0.0` - http://bosh.io/releases/github.com/logsearch/logsearch-boshrelease?version=202.0.0

**Comment by hannayurkevich** [comment]
@mrdavidlaing 

Thank you very much!
